packages/client/src/workflow-client.ts

```diff
@@ -447,6 +453,44 @@ export interface IntoHistoriesOptions {
   bufferLimit?: number;
 }
 
+const withStartWorkflowOperationResolve: unique symbol = Symbol();
+const withStartWorkflowOperationReject: unique symbol = Symbol();
+const withStartWorkflowOperationUsed: unique symbol = Symbol();
+
+/**
+ * Defines how to start a workflow when using {@link WorkflowClient.startUpdateWithStart} and
+ * {@link WorkflowClient.executeUpdateWithStart}. `workflowIdConflictPolicy` is required in the options.
+ *
+ * @experimental
+ */
+export class WithStartWorkflowOperation<T extends Workflow> {
+  public [withStartWorkflowOperationUsed]: boolean = false;
+  public [withStartWorkflowOperationResolve]: ((handle: WorkflowHandle<T>) => void) | undefined = undefined;
+  public [withStartWorkflowOperationReject]: ((error: any) => void) | undefined = undefined;
+  private workflowHandlePromise: Promise<WorkflowHandle<T>>;
+
+  private constructor(
+    public workflowTypeOrFunc: string | T,
```

Comment:
Sorry for the uncertainty. Let's revert to have a public constructor.

Reply:
Done